### PR TITLE
virtualjaguar remove unused eclass, -egl, src_configure return and install desktop icon+file.

### PR DESCRIPTION
**Pull request by damex**
short description of the changes:
1. eutils is not used here. at'all.
2. with egl enabled virtualjaguar does not provide any graphics output.
3. it does repeat eqmake\* on src_configure and fails to build after.
4. it does not install bundled desktop icon with desktop file.

### CODE REVIEW

File: games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild [hunk 1]
```diff
@@ -30,6 +30,10 @@ src_prepare() {
 	eqmake4 virtualjaguar.pro -o makefile-qt
 }
 
+src_configure() {
+	return
```
Comment (mgorny) [inline games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild]: If you want to no-op, something, `:`.

File: games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # $Header: /var/cvsroot/gentoo-x86/games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild,v 1.1 2014/10/15 03:50:52 mr_bones_ Exp $
 
 EAPI=5
-inherit eutils qt4-r2 games
+inherit qt4-r2 games
```
Comment (mgorny) [inline games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild]: The funny thing is…. are you sure it doesn't use eutils _now_?

File: games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild [hunk 3]
```diff
@@ -18,8 +18,8 @@ RDEPEND="media-libs/libsdl[joystick,opengl,sound,video]
 	sys-libs/zlib
 	virtual/opengl
 	dev-qt/qtcore:4
-	dev-qt/qtgui:4
-	dev-qt/qtopengl:4
+	dev-qt/qtgui:4[-egl]
+	dev-qt/qtopengl:4[-egl]
```
Comment (mgorny) [inline games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild:22]: I'm not sure if we're supposed to depend on that directly. Let's ask someone…

File: games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild (hunk 2 above)
Comment (damex) [inline games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild]: actually "domenu" use it now, but eutils eclass automatically inherited with games eclass... and we actually do it twice in a row.

File: games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild (hunk 3 above)
Comment (junghans) [inline games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild:22]: The question is, does `dev-qt/qtgui` with `egl` enabled, trigger some problems? Or does `games-emulation/virtualjaguar` just make no use of `egl` in `dev-qt/qtgui`.

File: games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild [hunk 4]
```diff
@@ -18,8 +18,8 @@ RDEPEND="media-libs/libsdl[joystick,opengl,sound,video]
...
 	dev-libs/libcdio"
 DEPEND="${RDEPEND}
 	>=sys-devel/gcc-4.4"
```
Comment (junghans) [inline games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild:25]: That looks strange, too. Should we check the selected `gcc` in `src_pretend`?

File: games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild (hunk 2 above)
Comment (junghans) [inline games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild]: If the ebuild uses the eclass directly, keep it here! We can never be sure that future versions of `games.eclass` will still inherit `eutils.eclass`.

Comment (prometheanfire) [inline games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild]: :+1: to keep it

File: games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild (hunk 3 above)
Comment (mgorny) [inline games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild:22]: Well, it does :). But I'm not sure if that's virtualjaguar's fault or Qt/EGL/video card problem. @damex's getting some rubbish static on the screen, I'm getting:

```
QEglContext::swapBuffers(): "Bad surface (0x300D)" 
QEglContext::swapBuffers(): "Bad surface (0x300D)" 
QEglContext::swapBuffers(): "Bad surface (0x300D)" 
QEglContext::swapBuffers(): "Bad surface (0x300D)" 
QEglContext::swapBuffers(): "Bad surface (0x300D)" 
```

File: games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild (hunk 4 above)
Comment (mgorny) [inline games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild:25]: Correct. Though I'm not sure if it's worth the effort for such an ancient version.

File: games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild (hunk 3 above)
Comment (damex) [inline games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild:22]: asked virtualjaguar developer.
<damex|work> is it virtualjaguar trouble with egl?
<shamus397> yeah, there's something that needs to be patched to work with egl in the glwidget.h/cpp

Comment (mgorny) [inline games-emulation/virtualjaguar/virtualjaguar-2.1.2.ebuild:22]: Ok then :).

**Comment by mgorny** [comment]
So I'd say it looks fine for merging now. Normally I'd tell you to rebase and merge the changes into one commit but since we're merging that into CVS anyway that's not necessary. I'm going to check if my merging script works fine now :).

**Comment by mgorny** [comment]
Ok, one more request if you don't mind learning a bit more about ebuild development. Since after your changes ebuild starts installing additional files (which were missing before), please bump the revision (`ebump -c`) and leave the old revision alone. This will ensure that all users get the upgrade.

**Pull request closed without merging**